moveit_ros/moveit_servo/src/pose_tracking.cpp

```diff
@@ -137,63 +139,70 @@ PoseTrackingStatusCode PoseTracking::moveToPose(const Eigen::Vector3d& positiona
 void PoseTracking::readROSParams()
 {
   // Optional parameter sub-namespace specified in the launch file. All other parameters will be read from this namespace.
-  std::string parameter_ns;
-  ros::param::get("~parameter_ns", parameter_ns);
+  rclcpp::Parameter parameter_ns;
+  //std::string parameter_ns;
+  node_->get_parameter("~parameter_ns", parameter_ns);
 
   // If parameters have been loaded into sub-namespace within the node namespace, append the parameter namespace
   // to load the parameters correctly.
-  ros::NodeHandle nh = parameter_ns.empty() ? nh_ : ros::NodeHandle(nh_, parameter_ns);
+  rclcpp::Node::SharedPtr node;
+  //= node_->has_parameter("~parameter_ns") ? node_ : rclcpp::Node::SharedPtr(node_, parameter_ns.as_string());
+  if (node_->has_parameter("~parameter_ns"))
+    node = node_;
+  else
+    node->declare_parameter("~parameter_ns");
+    //node->set_parameter("~parameter_ns", parameter_ns.as_string());
 
   // Wait for ROS parameters to load
-  ros::Time begin = ros::Time::now();
-  while (ros::ok() && !nh.hasParam("planning_frame") && ((ros::Time::now() - begin).toSec() < ROS_STARTUP_WAIT))
+  rclcpp::Time begin = node->now();
+  while (rclcpp::ok() && !node->has_parameter("planning_frame") && ((node->now() - begin).to_chrono<std::chrono::duration<double>>().count() < ROS_STARTUP_WAIT))
   {
-    ROS_WARN_STREAM_NAMED(LOGNAME, "Waiting for parameter: "
-                                       << "planning_frame");
-    ros::Duration(0.1).sleep();
+    RCLCPP_WARN_STREAM(LOGGER, "Waiting for parameter: "
+                                   << "planning_frame");
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
   }
 
   std::size_t error = 0;
 
-  error += !rosparam_shortcuts::get(LOGNAME, nh, "planning_frame", planning_frame_);
-  error += !rosparam_shortcuts::get(LOGNAME, nh, "move_group_name", move_group_name_);
+  error += !PoseTracking::rosparam_shortcut_get(LOGGER, node, "planning_frame", planning_frame_);
+  error += !PoseTracking::rosparam_shortcut_get(LOGGER, node, "move_group_name", move_group_name_);
   if (!planning_scene_monitor_->getRobotModel()->hasJointModelGroup(move_group_name_))
   {
     ++error;
-    ROS_ERROR_STREAM_NAMED(LOGNAME, "Unable to find the specified joint model group: " << move_group_name_);
+    RCLCPP_ERROR_STREAM(LOGGER, "Unable to find the specified joint model group: " << move_group_name_);
   }
 
   double publish_period;
-  error += !rosparam_shortcuts::get(LOGNAME, nh, "publish_period", publish_period);
-  loop_rate_ = ros::Rate(1 / publish_period);
+  error += !PoseTracking::rosparam_shortcut_get(LOGGER, node, "publish_period", publish_period);
+  rclcpp::Rate loop_rate_(publish_period);
```

Comment:
I think double values are initialized as rate, so we should probably stick to using `1 / publish_period`.